high_card.rb

```diff
@@ -1,5 +1,78 @@
-# Use these two arrays to generate a deck of cards.
+require "pry"
+
+def build_deck
+
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
+deck = []
+
+#build deck of cards
+ranks.each_with_index do |rank, i|
+  suits.each do |suit|
+    deck << [rank, suit, i]
+  end
+end
+return deck.shuffle
+
+end
+
+#create players:
+def create_players
+  players=[]
+  while true
+    puts "Enter name of player or type 'start' to begin"
+    playerName = gets.chomp
+
+    if playerName == "start"
+      break
+    end
+      players << playerName
+    end
+  return players
+end
+
+# Deal/Map players into cards
+def deal
+  deck = build_deck
+    players = create_players
+    hands = []
+    puts "Dealing cards..."
+    players.each do |player|
+      hand = {
+        player: player, card: deck.pop
+      }
+      hands << hand
+    end
+    return hands
+  end
+
+#ROADBLOCK: Would there be a way to deal the cards and pause before determining the winner?
```

Comment:
could do either `gets.chomp` or http://stackoverflow.com/questions/1329967/tell-ruby-program-to-wait-some-amount-of-time